### PR TITLE
PE-4146: disable automatic sync when user is on the topup modal

### PR DESCRIPTION
**Pull request by thiagocarvalhodev**
--- Releases ---
Android release: https://appdistribution.firebase.google.com/testerapps/1:305132849030:android:6cf0cd5ec064fad3ffce07/releases/7jvrjtlnaap48
iOS release: https://appdistribution.firebase.google.com/testerapps/1:305132849030:ios:06ea33a95a2e0d42ffce07/releases/1emqn45sgg96o

### CODE REVIEW

File: lib/blocs/sync/sync_cubit.dart [hunk 1]
```diff
@@ -183,6 +187,11 @@ class SyncCubit extends Cubit<SyncState> {
   var ghostFolders = <FolderID, GhostFolder>{};
 
   Future<void> startSync({bool syncDeep = false}) async {
+    if (!_activityTracker.isSyncAllowed) {
+      logger.i('Activity tracker is not allowing sync');
```
Comment (karlprieb) [inline lib/blocs/sync/sync_cubit.dart:191]:
```suggestion
      logger.d('Activity tracker is not allowing sync');
```

**Review by karlprieb — DISMISSED** [review]
I'll approve so we can move one with Turbo, but I don't like the implementation. I was expecting something more generic that can track activities and decide if sync is allowed or not.
It's a good start, but too Turbo specific for now

File: lib/core/activity_tracker.dart [hunk 2]
```diff
@@ -0,0 +1,14 @@
+class ActivityTracker {
+  bool _isToppingUp = false;
+  // getters
+  bool get isToppingUp => _isToppingUp;
+
+  // setters
+  void setToppingUp(bool value) {
```
Comment (matibat) [inline lib/core/activity_tracker.dart:7]: The setter should be defined as so:
```suggestion
  set setToppingUp(bool value) {
```

File: lib/turbo/topup/views/topup_modal.dart [hunk 3]
```diff
@@ -54,6 +55,8 @@ void showTurboModal(BuildContext context, {Function()? onSuccess}) {
 
   initializeStripe(context.read<ConfigService>().config);
 
+  context.read<ActivityTracker>().setToppingUp(true);
```
Comment (matibat) [inline lib/turbo/topup/views/topup_modal.dart:58]: Will change if you replace the method approach with the setter approach
```suggestion
  context.read<ActivityTracker>().setToppingUp = true;
```

File: lib/turbo/topup/views/topup_modal.dart [hunk 4]
```diff
@@ -82,6 +85,8 @@ void showTurboModal(BuildContext context, {Function()? onSuccess}) {
       onSuccess?.call();
     }
 
+    context.read<ActivityTracker>().setToppingUp(false);
```
Comment (matibat) [inline lib/turbo/topup/views/topup_modal.dart:88]: Will change if you replace the method approach to the setter approach
```suggestion
    context.read<ActivityTracker>().setToppingUp = false;
```

**Review by matibat — COMMENTED** [review]
Left some comments. It's also lacking unit tests, and maybe bloc tests for sync (not sure where)

**Review by karlprieb — APPROVED** [review]

**Pull request merged**